Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -620,10 +621,37 @@ private static async Task PostAsync(string path, string requestId, string conten
                     using (var response = await client.PostAsync(path, payload))
                     {
                         statusCode = response.StatusCode;
+                        if (response.Content != null)
+                        {
+                            resContent = response.Content.ReadAsStringAsync().Result;
+                        }
+
                         response.EnsureSuccessStatusCode();
                     }
+
+                    if (path.Equals(Constants.UpdateDeployStatusPath) && resContent.Contains("Excessive SCM Site operation requests. Retry after 5 seconds"))

```

Comment:
String comparison doesn't look like a good idea. Isn't there a specific header that you can look at ? 
Also, wouldn't response.EnsureSuccessStatusCode() throw before you reach here?

Reply:
This one is a bit weird. @purva-vasudeo was looking into it. From what I could tell we were getting back 200 responses even though throttling was occurring. She could say more.

Reply:
Yes for some reason the content set by the throttling method was persisting but the statusCode was getting modified to 200 somewhere in the pipeline. Due to this EnsureSuccessStatusCode wasn't throwing an exception. @sanchitmehta and I were trying to figure out where this might be happening. We decided to go ahead with the string compare to unblock everyone for now.

Reply:
@shrishrirang The pipeline is setting the correct response code, but I believe it's modified in the native IIS module somewhere. We have added a WI to track this! We would move to using the status codes instead, this is just a work around to get the initial flow working.